tests/test_sandbox/test_sandbox_lifecycle.py

```
@@ -173,6 +173,29 @@ async def test_apply_cmd_executed(self):
+        """Diagnostics extract workspace.tar into /tmp/_nel_ws before reading the patch."""
+        apply_cmd = "APPLY_MARKER git apply /input/patch.diff"
+        ctx, mgr = self._make_ctx(apply_cmd=apply_cmd)
+        transfer = HostVolumeTransfer()
+        lc = StatelessSandbox(ctx, transfer)
+
+        await lc.setup()
+        await lc.get_agent_sandbox()
+        await lc.transition_to_verify("response", solver_modified=False)
+        verify_sb = await lc.get_verify_sandbox()
+
+        diag_cmds = [cmd for cmd in verify_sb._exec_log if "/tmp/_nel_ws/_nel_patch.diff" in cmd and "tar xf" in cmd]
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/tmp/_nel_ws/_nel_patch.diff"

(S108)